src/terminal/Grid.h

```diff
@@ -563,10 +563,24 @@ class Grid
     void rotateBuffersRight(LineCount count) noexcept { lines_.rotate_right(unbox<size_t>(count)); }
     // }}}
 
+    // {{{
+    void defineMaxHistoryLineCount_(LineCount _maxHistoryLineCount)
+    {
+        if(unbox<int>(_maxHistoryLineCount)==-1)
+        {
+            maxHistoryLineCount_ = LineCount::cast_from(0);
+            isInfinite_ = true;
+        }else {
+            maxHistoryLineCount_ = _maxHistoryLineCount;
+            isInfinite_ = false;
```

Comment:
While this works, this makes both variables dependant on each other.

Isn't it that either you have a maximum history line count or you have no limit ag all (infinite)? So with this in mind I think having both exclusive makes more sense to me, that is (something similar to):

```cpp
struct Infinite {};
using MaxHistoryLineCount = std::variant<LineCount, Infinite>;
```
Then make use of this strong type below and use `MaxHistoryLineCount` rather than `LineCount` for storing this value. That would also revert the required change in the constructor again (not needed).

The implication would be that you'd need to touch more code, all the lines that read from `maxHistoryLineCount_`, but wouldn't that be correct?

NB: Since sometimes `maxHistoryLineCount_` is being read from, even though it's meant to be infinite (optionally), I think you could achieve the same by reading the value `lines_.size()`, since this is pre-allocated anyways (with history line count + main page line couynt).

What do you think?

Reply:
p.s.: It would mean more coding (sorry for that), but I think that would make the intend much more clear and less error prone in future code changes due to less side data dependent variables one has to take care of.

Reply:
> While this works, this makes both variables dependant on each other.
> 
> Isn't it that either you have a maximum history line count or you have no limit ag all (infinite)? So with this in mind I think having both exclusive makes more sense to me, that is (something similar to):
> 
> ```c++
> struct Infinite {};
> using MaxHistoryLineCount = std::variant<LineCount, Infinite>;
> ```
> 
> Then make use of this strong type below and use `MaxHistoryLineCount` rather than `LineCount` for storing this value. That would also revert the required change in the constructor again (not needed).
> 
> The implication would be that you'd need to touch more code, all the lines that read from `maxHistoryLineCount_`, but wouldn't that be correct?
> 
> NB: Since sometimes `maxHistoryLineCount_` is being read from, even though it's meant to be infinite (optionally), I think you could achieve the same by reading the value `lines_.size()`, since this is pre-allocated anyways (with history line count + main page line couynt).
> 
> What do you think?

I've updated my implementation, now it is based on variant